.github/workflows/build.yml

```diff
@@ -0,0 +1,75 @@
+name: Test and Build Workflow
+# This workflow is triggered on pull requests to main branch
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    # Job name
+    name: Build Replication plugin
+    runs-on: ubuntu-latest
+    steps:
+      # This step uses the setup-java Github action: https://github.com/actions/setup-java
+      - name: Set Up JDK 14
+        uses: actions/setup-java@v1
+        with:
+          java-version: 14
+      # dependencies: OpenSearch
+      - name: Checkout OpenSearch
+        uses: actions/checkout@v2
+        with:
+          # TODO: Point to opensearch repo after the patch changes have been merged.
+          repository: 'saikaranam-amazon/OpenSearch'
+          #repository: 'opensearch-project/OpenSearch'
+          path: OpenSearch
+          ref: 'karanas-os-replication'
+      - name: Build OpenSearch
+        working-directory: ./OpenSearch
+        run: |
+          ./gradlew publishToMavenLocal -Dbuild.snapshot=false
```

Comment:
You should be building a snapshot here, remove `-Dbuild.snapshot=false`

Reply:
Done